src/dgemm_kernel.rs

```diff
@@ -263,19 +263,19 @@ unsafe fn kernel_x86_avx<MA>(k: usize, alpha: T, a: *const T, b: *const T,
         //
         // 0 1 2 3 (the original)
         // 1 0 3 2 (chosen because of _mm256_shuffle_pd)
-        // 2 3 0 1 (chosen because of _mm256_permute2f128_pd)
-        // 3 0 1 2 (chosen because of _mm256_shuffle_pd applied after _mm256_permute2f128_pd)
+        // 3 2 1 0 (chosen because of _mm256_permute2f128_pd)
+        // 2 3 0 1 (chosen because of _mm256_shuffle_pd)
         let b_1032 = _mm256_shuffle_pd(b_0123, b_0123, 0b0101);
 
         // Both packed 4-vectors are the same, so one could also perform
-        // the selection 0b0000_0001 or 0b0011_0010.
+        // the selection 0b0000_0001 or 0b0010_0001 or 0b0010_0011.
         // The confusing part is that of the lower 4 bits and upper 4 bits
         // only 2 bits are used in each. The same choice could have been
         // encoded in a nibble (4 bits) total, i.e. 0b1100, had the intrinsics
         // been defined differently. The highest bit in each nibble controls
         // zero-ing behaviour though.
-        // let b_2301 = _mm256_permute2f128_pd(b_0123, b_0123, 0b0011_0000);
-        // 0b0011_0000 = 0x30; makes it clearer which bits we are acting on.
+        // let b_3210 = _mm256_permute2f128_pd(b_1032, b_1032, 0b0000_0011);
+        // 0b0000_0011 = 0x03; makes it clearer which bits we are acting on.
```

Comment:
This comment just repeats the code below now so, can't it just be removed? And if binary literals are clearer, then we can replace 0x3 with 0b0011 in the code :)

When looking at this I was wondering wouldn't 

```rust
        let b_2301 = _mm256_permute2f128_pd(b_0123, b_0123, 0x03);
        let b_1032 = _mm256_shuffle_pd(b_0123, b_0123, 0b0101);
        let b_3210 = _mm256_shuffle_pd(b_2301, b_2301, 0b0101);
```

Be faster? Because previously we have shuf -> perm -> shuf dependency chain and the new one just has perm -> {shuf, shuf}. But in benchmarks I can't see much of a difference.

If you want to run benchmarks in the repo you can use something like this:

```
./benches/benchloop.py -s 128 -t f64 | tee /dev/stderr | xsv table
```

Benchloop outputs CSV on stdout so one can use a tool like xsv to format it into a table. Benchloop is used to run the benchmark while varying various variables (element type, size, other parameters)

Reply:
cc @SuperFluffy since it touches code you've once contributed

Reply:
> These last lines of the comment just repeat the code below now so, can't that part just be removed? And if binary literals are clearer, then we can replace 0x3 with 0b0011 in the code :)

In my opinion yes to both. I just wanted to make the smallest amount of changes initially and see if such minor changes are even welcome :)

> When looking at this I was wondering wouldn't
        let b_2301 = _mm256_permute2f128_pd(b_0123, b_0123, 0x03);
        let b_1032 = _mm256_shuffle_pd(b_0123, b_0123, 0b0101);
        let b_3210 = _mm256_shuffle_pd(b_2301, b_2301, 0b0101);
Be faster? Because previously we have shuf -> perm -> shuf dependency chain and the new one just has perm -> {shuf, shuf}. But in benchmarks I can't see much of a difference.

While reading the code I also wondered that. Theoretically it should be better in my understanding, but I am by no means an expert. I just ran the benchmarks you mentioned and also didn't see any significant difference. If the change is desired I would be happy to include it in this PR or make a new one.

Reply:
> These last lines of the comment just repeat the code below now so, can't that part just be removed? And if binary literals are clearer, then we can replace 0x3 with 0b0011 in the code :)

Just pushed these changes.

Reply:
Awesome, thank you @Tastaturtaste for going through these.

If all tests pass I think whoever spent the time thinking through this code is most qualified to make changes to it.

Reply:
Sounds like a good way to go about it :slightly_smiling_face: